include/kframework/lemmas/int-simplification.k

```diff
@@ -115,13 +115,27 @@ module INT-SIMPLIFICATION-COMMON
   // max, min
   // ###########################################################################
 
-    rule [minint-left]: minInt(A, B) => A requires A <=Int B [simplification]
+    rule [minint-left]:  minInt(A, B) => A requires A <=Int B [simplification]
+    rule [minint-right]: minInt(A, B) => B requires B <=Int A [simplification]
 
-    rule [minint-lt]:  minInt(A, B) <Int C  => true requires A  <Int C  orBool B  <Int C [simplification]
-    rule [minint-gt]:  A  <Int minInt(B, C) => true requires A  <Int B andBool A  <Int C [simplification]
-    rule [minint-geq]: A <=Int minInt(B, C) => true requires A <=Int B andBool A <=Int C [simplification]
+    rule [minint-lt]:  minInt(A, B)  <Int C => A  <Int C  orBool B  <Int C [simplification]
+    rule [minint-leq]: minInt(A, B) <=Int C => A <=Int C  orBool B <=Int C [simplification]
+    rule [minint-gt]:  A  <Int minInt(B, C) => A  <Int B andBool A  <Int C [simplification]
+    rule [minint-geq]: A <=Int minInt(B, C) => A <=Int B andBool A <=Int C [simplification]
```

Comment:
Same here, can we split each and have lemmas that only go to `true/false` (or directly to one of the branches)?

In the case that the resulting RHS is simplifiable, this expression will go directl yto `true/false`, or to one of the branches anyway. So having it go directly there happens in 1 rewrite instead of 2. In the case that the RHS here is not simplifiable further, now we have an expression with lower structure, and we don't know that these came from a maxInt/minInt.

Reply:
Same as above.